File: src/main/java/org/jetbrains/space/jenkins/scm/SpaceSCMSourceRequest.java
package org.jetbrains.space.jenkins.scm;

import hudson.model.TaskListener;
import jenkins.scm.api.SCMSource;
import jenkins.scm.api.trait.SCMSourceRequest;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;

import static java.util.Objects.requireNonNull;

public class SpaceSCMSourceRequest extends SCMSourceRequest {

    private final Collection<SpaceSCMHeadDiscoveryHandler> discoveryHandlers;

    protected SpaceSCMSourceRequest(@NotNull SCMSource source, @NotNull SpaceSCMSourceContext context, TaskListener listener) {
        super(source, context, listener);
        this.discoveryHandlers = requireNonNull(context.getDiscoveryHandlers(), "discoveryHandlers");
    }

    public Collection<SpaceSCMHeadDiscoveryHandler> getDiscoveryHandlers() {
        return discoveryHandlers;
    }

}

File: src/main/java/org/jetbrains/space/jenkins/scm/SpaceSCMSourceRequest.kt
package org.jetbrains.space.jenkins.scm

import hudson.model.TaskListener
import jenkins.scm.api.SCMSource
import jenkins.scm.api.trait.SCMSourceRequest

open class SpaceSCMSourceRequest(source: SCMSource, context: SpaceSCMSourceContext, listener: TaskListener?) : SCMSourceRequest(source, context, listener) {
    val discoveryHandlers = context.discoveryHandlers
}
